broker-daemon/interchain-router/get-preimage.js

```diff
@@ -83,6 +83,22 @@ async function getPreimage ({ params, send, onCancel, onError, ordersByHash, eng
   const { inboundSymbol, inboundFillAmount, outboundSymbol, outboundFillAmount, takerAddress } = order
   const [ expectedSymbol, actualSymbol, expectedAmount, actualAmount ] = [ inboundSymbol, symbol, inboundFillAmount, amount ]
 
+  const outboundEngine = engines.get(outboundSymbol)
+  if (!outboundEngine) {
+    const err = `No engine available for ${outboundSymbol}`
+    logger.error(err)
+    return send({ permanentError: err })
+  }
+
+  if (await outboundEngine.isPaymentPendingOrComplete(swapHash)) {
+    const { paymentPreimage, permanentError } = await outboundEngine.getPaymentPreimage(swapHash)
+    if (permanentError) {
```

Comment:
It looks like this if statement isnt needed. If one of the params is undefined, it wont be added to the response.

We can simple do `return send({ paymentPreimage, permanentError })`